src/NetMQ/NetMQPoller.cs

```diff
@@ -120,25 +120,63 @@ protected override void QueueTask(Task task)
             m_tasksQueue.Enqueue(task);
         }
 
+        internal Task RunAsync([NotNull] Action action)
+        {
+            Task t;
+
+            if (!IsRunning || CanExecuteTaskInline)
+            {
+                action();
+
+                t = FromResult<object>(null);
+            }
+            else
+            {
+                t = Task.Factory.StartNew(action);
```

Comment:
The `Run` method's purpose was to execute `action` on the poller's thread. It did this by providing `this` as the task scheduler.

This new `RunAsync` method does not guarantee the action runs on the poller's thread.

Reply:
oops.  You're right.  Will fix.